pytest_fixtures/component/satellite_auth.py

```diff
@@ -226,64 +227,78 @@ def ldap_auth_source(request, module_org, module_location, ad_data, ipa_data, op
     yield ldap_data, auth_source
 
 
-@pytest.fixture(scope='session')
-def enroll_configure_rhsso_external_auth():
+@pytest.fixture
+def auth_data(request, ad_data, ipa_data):
+    auth_type = request.param.lower()
+    if 'ad' in auth_type:
+        ad_data = ad_data('2019') if '2019' in auth_type else ad_data()
+        ad_data['server_type'] = LDAP_SERVER_TYPE['UI']['ad']
+        ad_data['attr_login'] = LDAP_ATTR['login_ad']
+        ad_data['auth_type'] = auth_type
+        return ad_data
+    elif auth_type == 'ipa':
+        ipa_data['server_type'] = LDAP_SERVER_TYPE['UI']['ipa']
+        ipa_data['attr_login'] = LDAP_ATTR['login']
+        ipa_data['auth_type'] = auth_type
+        return ipa_data
+
+
+@pytest.fixture(scope='module')
+def enroll_configure_rhsso_external_auth(module_target_sat):
     """Enroll the Satellite6 Server to an RHSSO Server."""
-    run_command(
-        cmd='yum -y --disableplugin=foreman-protector install '
+    module_target_sat.execute(
+        'yum -y --disableplugin=foreman-protector install '
         'mod_auth_openidc keycloak-httpd-client-install'
     )
-    run_command(
-        cmd=f'echo {settings.rhsso.password} | keycloak-httpd-client-install --app-name foreman-openidc \
+    module_target_sat.execute(
+        f'echo {settings.rhsso.rhsso_password} | keycloak-httpd-client-install --app-name foreman-openidc \
                 --keycloak-server-url {settings.rhsso.host_url} \
                 --keycloak-admin-username "admin" \
                 --keycloak-realm "{settings.rhsso.realm}" \
                 --keycloak-admin-realm master \
                 --keycloak-auth-role root-admin -t openidc -l /users/extlogin --force'
     )
-    run_command(
-        cmd=f'satellite-installer --foreman-keycloak true '
+    module_target_sat.execute(
+        f'satellite-installer --foreman-keycloak true '
         f"--foreman-keycloak-app-name 'foreman-openidc' "
         f"--foreman-keycloak-realm '{settings.rhsso.realm}' ",
         timeout=1000000,
     )
-    run_command(cmd="systemctl restart httpd")
+    module_target_sat.execute('systemctl restart httpd')
```

Comment:
Use f-m CLI's instead of systemctl
```suggestion
    module_target_sat.cli.Service.restart(options={'only': 'httpd'})
```

Reply:
The only thing is holding me the documentation here https://access.redhat.com/documentation/en-us/red_hat_satellite/6.11/html/administering_red_hat_satellite/configuring_external_authentication_admin#Configuring_Keycloak_Authentication_with_TOTP_keycloak-totp. It is mentioned that we need to use `systemctl restart httpd`. Just wanted to follow the steps mentioned in the docs. I know both would do the same thing.

Reply:
Sure, it is just a suggestion, you can stick with systemctl, but I believe docs should be updated to use f-m everywhere to manage services. I'll open a docs BZ for this after discussing this with devs.